src/rgw/rgw_cache.cc

```diff
@@ -124,14 +124,12 @@ void ObjectCache::put(const string& name, ObjectCacheInfo& info, rgw_cache_entry
 
   ldout(cct, 10) << "cache put: name=" << name << " info.flags=0x"
                  << std::hex << info.flags << std::dec << dendl;
-  auto iter = cache_map.find(name);
-  if (iter == cache_map.end()) {
-    ObjectCacheEntry entry;
+
+  auto [iter, inserted] = cache_map.emplace(name, ObjectCacheEntry{});
```

Comment:
does this construct an ObjectCacheEntry that may be discarded?

Reply:
yes. the constructor doesn't do anything expensive though, now that `time_added` is no longer initialized inline

Reply:
You could use C++17s fancy, new `try_emplace` to avoid the constructor call if the object already exists.